test/CMakeLists.txt

```diff
@@ -113,6 +113,31 @@ function(get_test_dependencies SDK result_var_name)
   set("${result_var_name}" "${deps}" PARENT_SCOPE)
 endfunction()
 
+function (write_component_test_script test_bin_dir component_name)
+  string(REPLACE ";" " " args "${LIT_ARGS}")
+  file(
+    WRITE
+    ${test_bin_dir}/check-swift-${component_name}.sh
+    "${PYTHON_EXECUTABLE} ${LIT} \\
+    ${args} \\
+    ${test_bin_dir}/ \\
+    --filter=${component_name}\n")
```

Comment:
This really should just be a custom command that has an associated target IMO rather than a script - that would allow you to do `ninja check-swift-<component name>` which would just run the script (without having to write out the oneline wrapper script and make this something that can be used on any target (which I think is also important so that @CodaFi could reproduce this on say Windows).

Reply:
The reason why we didn't make them ninja targets is because we thought people expect the build system to do a consistent build. Whereas here we are providing a way to run tests in a non-hermetic way. Specifically, the bottleneck is often in rebuilding the standard library. The workflow we are trying to keep going is:

```
ninja bin/swift
# Saving some time by not rebuilding the standard library and running lit directly!
lit.py test/SomeComponent
```

Reply:
Id argue that `check-*` should just be aliases which run lit :)

Reply:
Do you mean all of the `check-*` targets, including `check-all`? Today they are not just running lit -- they do a consistent build first, and then run tests. Changing that would be a much bigger change than what we are trying to do.